test/fuzz/tests/crypto_merkle_test.go

```diff
@@ -0,0 +1,47 @@
+//go:build gofuzz || go1.18
+
+package tests
+
+import (
+	"bytes"
+	"testing"
+
+	tmcrypto "github.com/tendermint/tendermint/crypto"
+	"github.com/tendermint/tendermint/crypto/merkle"
+	tmrand "github.com/tendermint/tendermint/libs/rand"
+)
+
+func FuzzProofsFromByteSlices(f *testing.F) {
+	seeds := [][]byte{
+		nil,
+		{},
+	}
+
+	// [][]byte isn't supported as an input hence add []byte values.
+	var fakeSeparator = []byte("***FAKE_MARKER***")
+
+	for i := 0; i < 10; i++ {
+		total := 100
+		seed := make([][]byte, total)
+		for j := 0; j < total; j++ {
+			seed[j] = tmrand.Bytes(tmcrypto.HashSize)
+		}
+		seeds = append(seeds, bytes.Join(seed, fakeSeparator))
+	}
+
+	for _, seed := range seeds {
+		f.Add(seed)
+	}
+
+	f.Fuzz(func(t *testing.T, joinedInput []byte) {
```

Comment:
Usually, fuzzing is used to tease out corner cases where a parser/decoder panics because of unexpected input. So while you can use `f.Errorf` or similar to flag a failing fuzz test, this is the usual pattern.

Besides, OSS-Fuzz [doesn't support](https://github.com/golang/go/issues/50192) native Go 1.18 fuzzing yet and [will ignore](https://github.com/AdamKorcz/go-118-fuzz-build/blob/7bed8f92ea820f668efab099ec774d548f1873a8/utils/test_helpers.go#L19) `f.Errorf` and friends.